pkg/controller/provider/container/vsphere/model.go

```diff
@@ -376,14 +376,59 @@ func (v *HostAdapter) Apply(u types.ObjectUpdate) {
 					v.model.HostScsiDisks = nil
 					for _, iScsiLun := range array.ScsiLun {
 						hostScsiDisk := model.HostScsiDisk{}
-						scsiLun := iScsiLun.GetScsiLun()
-						hostScsiDisk.CanonicalName = scsiLun.CanonicalName
-						hostScsiDisk.Vendor = strings.TrimSpace(scsiLun.Vendor)
-						v.model.HostScsiDisks = append(v.model.HostScsiDisks, hostScsiDisk)
+						if disk, ok := iScsiLun.(*types.HostScsiDisk); ok {
+							hostScsiDisk.CanonicalName = disk.CanonicalName
+							hostScsiDisk.Vendor = strings.TrimSpace(disk.Vendor)
+							hostScsiDisk.Model = strings.TrimSpace(disk.Vendor)
+							hostScsiDisk.Key = strings.TrimSpace(disk.Key)
+							v.model.HostScsiDisks = append(v.model.HostScsiDisks, hostScsiDisk)
+						}
 					}
 				}
 			case fAdvancedOption:
 				v.model.AdvancedOptions = v.Ref(p.Val)
+			case fHostBusAdapter:
+				if array, cast := p.Val.(types.ArrayOfHostHostBusAdapter); cast {
+					for _, hba := range array.HostHostBusAdapter {
+						hbaDiskInfo := model.HbaDiskInfo{}
+						protocolType := model.ProtocolUnknown
+						switch hba.(type) {
+						case *types.HostFibreChannelHba:
+							protocolType = model.ProtocolFibreChannel
+						case *types.HostFibreChannelOverEthernetHba:
+							protocolType = model.ProtocolFCoE
+						case *types.HostInternetScsiHba:
+							protocolType = model.ProtocolISCSI
+						case *types.HostParallelScsiHba:
+							protocolType = model.ProtocolSCSI
+						case *types.HostSerialAttachedHba:
+							protocolType = model.ProtocolSAS
+						case *types.HostPcieHba:
+							protocolType = model.ProtocolPCIe
+						case *types.HostRdmaHba:
+							protocolType = model.ProtocolRDMA
+						case *types.HostTcpHba:
+							protocolType = model.ProtocolTCP
+						}
+						hostHba := hba.GetHostHostBusAdapter()
+						hbaDiskInfo.Device = hostHba.Device
+						hbaDiskInfo.Model = hostHba.Model
+						hbaDiskInfo.Key = hostHba.Key
+						hbaDiskInfo.Protocol = string(protocolType)
+						v.model.HbaDiskInfo = append(v.model.HbaDiskInfo, hbaDiskInfo)
+					}
+				}
+			case fScsiTopology:
+				if array, cast := p.Val.(types.ArrayOfHostScsiTopologyInterface); cast {
+					for _, scsiTopologyInterface := range array.HostScsiTopologyInterface {
+						hostScsiTopology := model.HostScsiTopology{}
+						hostScsiTopology.HbaKey = scsiTopologyInterface.Key
+						for _, scsiTopologyTarget := range scsiTopologyInterface.Target {
+							hostScsiTopology.ScsiDiskKeys = append(hostScsiTopology.ScsiDiskKeys, scsiTopologyTarget.Key)
```

Comment:
IMHO you need here `lun.key` not `key`. But unsure how you plan to do the mappings.

Reply:
Yeah, I think you are comparing two non-comparable items. I think you need to fetch `lun.key`

Reply:
You right, I. used the wrong value here
But I think it sgould be lun. scsiLun (not key)
I used the first lun only (lun[0]) as its seems there is only one in this list and anyway it should be enough

Reply:
The work on the migration-planner and testing it still in progress

Reply:
The change was validated with migartion-planner cahnges:
https://github.com/kubev2v/migration-planner/pull/292
The data was collected as expected
@machacekondra @mnecas 
Can you please have the final review and merge?